src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Android/StorePal.AndroidKeyStore.cs

```diff
@@ -0,0 +1,149 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal sealed partial class StorePal
+    {
+        private sealed class AndroidKeyStore : IStorePal
+        {
+            private readonly bool _readOnly;
+            private readonly SafeX509StoreHandle _keyStoreHandle;
+
+            public static AndroidKeyStore OpenDefault(OpenFlags openFlags)
+            {
+                SafeX509StoreHandle store = Interop.AndroidCrypto.X509StoreOpenDefault();
+                if (store.IsInvalid)
+                    throw new CryptographicException();
```

Comment:
```suggestion
                if (store.IsInvalid)
                {
                    store.Dispose();
                    throw new CryptographicException();
                }
```

No reason to send it off to the finalizer queue.